packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,46 +7,88 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
+import {TDateRange} from './types';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	dateRange: TDateRange;
 	disabled?: boolean;
-	onClick: (date: Date) => void;
+	enableDateRange?: boolean;
+	onClick: (date: Date, enableDateRange?: boolean) => void;
+}
+
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, dirtyInterval: IInterval) {
```

Comment:
Mentioned this function on the PR description here: https://github.com/liferay/clay/pull/4008#issue-609258062